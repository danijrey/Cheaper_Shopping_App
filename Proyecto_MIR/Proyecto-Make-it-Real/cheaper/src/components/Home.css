.App {
	margin: 0;
	padding: 0;
}
.Home{
	width: 100%;
}
.NavBar-Container {
	display: flex;
	justify-content: space-around;
	text-align: center;
	background-color: #e0c308;
	width: 100vw;
}
.NavBar-Parts {
	width: 100%;
	height: 45px;
	display: flex;
	justify-content: space-around;
	align-items: center;

}

.NavBar-Logo-Home {
	width: 100px;
	height: 45px;
	text-align: center;
	margin:0px
}

.NavBar-Icons {
	color: whitesmoke;
	cursor: pointer;
	font-size: 20px;
	margin: 5px;
}

.NavBusqueda-Home {
	display: flex;
	justify-content: center;
	align-items: center;
	margin-top: 5%;
}


