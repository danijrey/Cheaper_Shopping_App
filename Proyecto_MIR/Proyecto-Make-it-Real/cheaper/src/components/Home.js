import React from "react";
import "./Home.css";
import Slider from "../components/slider";
import ProductListHome from "../components/productListHome.js";
import CreateProduct from "../components/createProduct.js";
import CreateBranch from "../components/createBranch.js";
import NavBar from './NavBar.js'


class Home extends React.Component {

  render() {

    return (
      <div className="Home">

        <NavBar />
        <CreateProduct />
        <CreateBranch />
        <Slider />
        <ProductListHome />

      </div>
    );
  }
}

export default Home;