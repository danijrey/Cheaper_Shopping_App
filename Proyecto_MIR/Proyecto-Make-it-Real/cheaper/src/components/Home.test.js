import React from 'react';
import { render, getByText } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Home from './Home';


describe('Home', () => {
  it('should render correctly', () => {
    const { container } = render(
      <MemoryRouter>
        <Home />
      </MemoryRouter>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
  it('Should render elements with the following Id = "HomeProvider","ProductList","Slider","NewBranchLink","NewProductLink" ', () => {
    const { getByTestId } = render(
      <MemoryRouter>
        <Home />
      </MemoryRouter>
    );
    const links = getByTestId('HomeProvider','ProductList', 'Slider', 'NewBranchLink', 'NewProductLink');
    expect(links).toBeInTheDocument();
  });
  it('Should render elemente with the following placeholder text= "búsqueda", "Ubicación"', () => {
    const { getByPlaceholderText } = render(
      <MemoryRouter>
        <Home />
      </MemoryRouter>
    );
    const placeholders = getByPlaceholderText(/búsqueda/, /Ubicación/);
    expect(placeholders).toBeInTheDocument();
  });
});