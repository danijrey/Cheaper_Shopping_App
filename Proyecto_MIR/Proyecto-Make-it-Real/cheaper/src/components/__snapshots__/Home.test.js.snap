// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render correctly 1`] = `
<div
  class="Home"
  data-testid="HomeProvider"
>
  <header
    class="toolbar"
  >
    <nav
      class="toolbar-navigation"
    >
      <div
        class="toolbar-toggle-button"
      >
        <button
          class="toggle-button"
        >
          <div
            class="toggle-button_line"
          />
          <div
            class="toggle-button_line"
          />
          <div
            class="toggle-button_line"
          />
        </button>
      </div>
      <img
        alt="Logo"
        class="toolbar-logo"
        src="logo.png"
      />
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-shopping-cart fa-w-18 NavBar-Icons"
        data-icon="shopping-cart"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 576 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M528.12 301.319l47.273-208C578.806 78.301 567.391 64 551.99 64H159.208l-9.166-44.81C147.758 8.021 137.93 0 126.529 0H24C10.745 0 0 10.745 0 24v16c0 13.255 10.745 24 24 24h69.883l70.248 343.435C147.325 417.1 136 435.222 136 456c0 30.928 25.072 56 56 56s56-25.072 56-56c0-15.674-6.447-29.835-16.824-40h209.647C430.447 426.165 424 440.326 424 456c0 30.928 25.072 56 56 56s56-25.072 56-56c0-22.172-12.888-41.332-31.579-50.405l5.517-24.276c3.413-15.018-8.002-29.319-23.403-29.319H218.117l-6.545-32h293.145c11.206 0 20.92-7.754 23.403-18.681z"
          fill="currentColor"
        />
      </svg>
    </nav>
  </header>
  <nav
    class="side-drawer"
  >
    <ul>
      <li>
        <a
          href="/"
        >
          Mi perfil
        </a>
      </li>
      <li>
        <a
          href="/products/all"
        >
          Mis productos
        </a>
      </li>
      <li>
        <a
          href="/home"
        >
          Inicio
        </a>
      </li>
      <li>
        <a
          href="/"
        >
          Configuracion
        </a>
      </li>
      <li>
        <a
          href="/providers/:id/branch/create"
        >
          Crear sucursal
        </a>
      </li>
      <img
        alt="Logo"
        class="side-drawer-logo"
        src="logo.png"
      />
    </ul>
  </nav>
  <div
    class="NavBusqueda-Home"
    data-testid="search"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-search fa-w-16 NavBusqueda-Icon"
      data-icon="search"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
        fill="currentColor"
      />
    </svg>
    <input
      class="NavBusqueda-Busqueda"
      placeholder=" Realizar una búsqueda."
      type="text"
    />
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-camera fa-w-16 NavBusqueda-Icon"
      data-icon="camera"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M512 144v288c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V144c0-26.5 21.5-48 48-48h88l12.3-32.9c7-18.7 24.9-31.1 44.9-31.1h125.5c20 0 37.9 12.4 44.9 31.1L376 96h88c26.5 0 48 21.5 48 48zM376 288c0-66.2-53.8-120-120-120s-120 53.8-120 120 53.8 120 120 120 120-53.8 120-120zm-32 0c0 48.5-39.5 88-88 88s-88-39.5-88-88 39.5-88 88-88 88 39.5 88 88z"
        fill="currentColor"
      />
    </svg>
  </div>
  <div
    class="Ubicacion"
    data-testid="location"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-map-marker-alt fa-w-12 Ubicacion-Icon"
      data-icon="map-marker-alt"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 384 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M172.268 501.67C26.97 291.031 0 269.413 0 192 0 85.961 85.961 0 192 0s192 85.961 192 192c0 77.413-26.97 99.031-172.268 309.67-9.535 13.774-29.93 13.773-39.464 0zM192 272c44.183 0 80-35.817 80-80s-35.817-80-80-80-80 35.817-80 80 35.817 80 80 80z"
        fill="currentColor"
      />
    </svg>
    <input
      class="Ubicacion-Busqueda"
      placeholder=" Ubicación."
      type="text"
    />
  </div>
  <span
    class="span-createProduct"
  >
    <a
      class="btn-createProduct"
      data-testid="NewProductLink"
      href="/products/create"
    >
      Crear Producto
    </a>
  </span>
  <span
    class="span-createBranch"
  >
    <a
      class="btn-createBranch"
      data-testid="NewBranchLink"
      href="/providers/:id/branch/create"
    >
      Crear Sucursal
    </a>
  </span>
  <div
    class="container-slider carousel slide"
    data-testid="Slider"
  >
    <ol
      class="carousel-indicators"
    >
      <li
        class="active"
      />
      <li />
      <li />
      <li />
      <li />
    </ol>
    <div
      class="carousel-inner"
    >
      <div
        class="active carousel-item"
      >
        <img
          alt="First slide"
          class="Food"
          src="food.png"
        />
        <div
          class="carousel-caption"
        >
          <h3
            class="H3-Food"
          >
            Descuentos hasta del 40%
          </h3>
          <p
            class="P-Food"
          >
            Pagando con tu tarjeta Éxito.
          </p>
        </div>
      </div>
      <div
        class="carousel-item"
      >
        <img
          alt="Third slide"
          class="Fruver"
          src="fruver.png"
        />
        <div
          class="carousel-caption"
        >
          <h3
            class="H3-Fruver"
          >
            Frutas siempre frsecas
          </h3>
          <p
            class="P-Fruver"
          >
            Aprovecha, tenemos promociones todos los Jueves
          </p>
        </div>
      </div>
      <div
        class="carousel-item"
      >
        <img
          alt="Third slide"
          class="Pan"
          src="pan.png"
        />
        <div
          class="carousel-caption"
        >
          <h3
            class="H3-Pan"
          >
            Tenemos sorpresas para ti
          </h3>
          <p
            class="P-Pan"
          >
            Solo en Jumbo, obtienes el 50%.Todos los Sábados
          </p>
        </div>
      </div>
      <div
        class="carousel-item"
      >
        <img
          alt="First slide"
          class="Chocolate"
          src="chocolate.png"
        />
        <div
          class="carousel-caption"
        >
          <h3
            class="H3-Chocolate"
          >
            Lo que más te gusta
          </h3>
          <p
            class="P-Chocolate"
          >
            Aprovecha la oferta en chocolates Suizos.
          </p>
        </div>
      </div>
      <div
        class="carousel-item"
      >
        <img
          alt="First slide"
          class="Carne"
          src="CarneSlider.jpg"
        />
        <div
          class="carousel-caption"
        >
          <h3
            class="H3-Carne"
          >
            25% De descuento
          </h3>
          <p
            class="P-Carne"
          >
            Todos los Martes en la Vaquita
          </p>
        </div>
      </div>
    </div>
    <a
      class="carousel-control-prev"
      href="#"
      role="button"
    >
      <span
        aria-hidden="true"
        class="carousel-control-prev-icon"
      />
      <span
        class="sr-only"
      >
        Previous
      </span>
    </a>
    <a
      class="carousel-control-next"
      href="#"
      role="button"
    >
      <span
        aria-hidden="true"
        class="carousel-control-next-icon"
      />
      <span
        class="sr-only"
      >
        Next
      </span>
    </a>
  </div>
  <div
    class="App"
  >
    <div
      class="gridContainer"
      data-testid="ProductList"
    />
  </div>
</div>
`;
